git_machete/cmd.py

```diff
@@ -1515,6 +1516,84 @@ def flush_caches(self) -> None:
         self.__branch_defs_by_sha_in_reflog = None
         self.__git.flush_caches()
 
+    def checkout_github_pr(self, pr_no: int) -> None:
+        org: str
+        repo: str
+        remote: str
+        remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        debug('checkout_github_pr', f'organization is {org}, repository is {repo}')
+
+        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
+        pr: GitHubPullRequest = check_pr_already_created_by_number(pr_no, all_prs)
+
+        if not pr:
+            raise MacheteException(f"PR number {pr_no} is not found in repository {repo}")
+        debug('checkout_github_pr', f'found {pr}')
+
+        current_branch: Optional[str] = self.__git.get_currently_checked_out_branch_or_none()
+        if current_branch == 'PR_' + str(pr.number):
+            self.__git.checkout(pr.head)
+
+        self.__git.fetch_pull_request(remote, pr.number)
+        debug('checkout_github_pr', f'fetched PR into branch PR_{pr}')
+
+        self.__git.fetch_remote(remote)
+
+        path: List[str] = self.get_path(remote, pr.head)
+        upstream_prs: List[GitHubPullRequest] = []
+        if len(path) > 1:
+            for index, branch in enumerate(path):
+                try:
+                    possible_pr: GitHubPullRequest = GitHubPullRequest(0, '', path[index + 1], branch, '')
+                    pr_found = check_pr_already_created(possible_pr, all_prs)
+                    if pr_found and pr_found != pr and pr_found.base != branch:
+                        upstream_prs.append(pr_found)
+                except IndexError:
+                    break
+
+        if upstream_prs:
+            reversed_path = path[::-1]  # need to add from root do down
+            for index, branch in enumerate(reversed_path):
+                if branch not in self.managed_branches:
+                    self.__cli_opts.opt_onto = reversed_path[index - 1]
+                    self.add(branch)
+
+            self.sync_annotations_to_github_prs()
+
+        self.__git.checkout(f'PR_{pr.number}')
+        print(fmt(f"Pull request #{pr.number} checked out at local branch PR_{pr.number}"))
+
+    def get_path(self, remote: str, last_branch: str) -> List[str]:
```

Comment:
Hmmm I didn't exactly investigate the logic here... but as discussed on a call yesterday, I wouldn't expect checking commit timestamps/reflogs etc. in finding out how the path up to an already machete-tracked branch should look like 🤔

Reply:
I'd rather completely rely on Github PR base branches for that purpose! See my PR comment about restoring the 3 branches based on their Github PRs

Reply:
Also, pls note that there is already `git machete discover` that tries to infer the parent for each branch based on things like reflog/latest checkout times etc.... although after dunno-how-many rounds of improvements it still ~sucks~ is far from perfection... actually, I've thought of `git machete github checkout-pr` as a more reliable alternative of sorts (at least for the case when there're already PRs in Github for the branches in question)

Reply:
it worked nice, until reflogs are available :/ which is rather rare. right now search is done by chain if pull requests upwards, so the code is not that complicated anymore.